### PR TITLE
prefers color scheme

### PR DESCRIPTION
**Pull request by romainmenke**
> - Breaking : removed old CDN urls
> - Breaking : remove `color-depth` queries fallback
> - Breaking : remove 'no-preference' support as this was dropped from the spec
> - Breaking : remove old global object

- cleaned up old and deprecated parts of the plugin
- converted the PostCSS plugin to typescript
- added browser tests

All docs have also been moved to a single [readme](https://github.com/csstools/postcss-plugins/tree/prefers-color-scheme--compassionate-alpine-dachsbracke-f5f77b6255/plugins/css-prefers-color-scheme#browser)

This was previously split up between browser and non-browser docs.
I hope that documenting it all together also makes it very clear how it all works.

------

_the individual issues have more details_

- fixes : https://github.com/csstools/postcss-plugins/issues/441
- fixes : https://github.com/csstools/postcss-plugins/issues/444
- fixes : https://github.com/csstools/postcss-plugins/issues/448
- fixes : https://github.com/csstools/postcss-plugins/issues/450

### CODE REVIEW

File: plugins/css-prefers-color-scheme/package.json [hunk 1]
```diff
@@ -35,23 +47,21 @@
 		"CHANGELOG.md",
 		"LICENSE.md",
 		"README.md",
-		"browser.js",
-		"browser.min.js",
 		"dist"
 	],
 	"peerDependencies": {
-		"postcss": "^8.4"
+		"postcss": "^8.3"
```
Comment (Antonio-Laguna) [inline plugins/css-prefers-color-scheme/package.json:53]: Any specific reason for this? Or is it just automatic change?

**Review by Antonio-Laguna — APPROVED** [review]
Nice work!

I started this page as a stub for these changes, https://github.com/csstools/postcss-plugins/wiki/PostCSS-Preset-Env-8

It might be worth to both, point to that on the CHANGELOG and also add any info on a per plugin basis. In this case, perhaps it might be even worth to point to the specific spec we're referring to on that wiki page which should be https://www.w3.org/TR/2021/WD-mediaqueries-5-20211218/#prefers-color-scheme (in case it changes again, though unlikely)

File: plugins/css-prefers-color-scheme/test/_browser.mjs [hunk 2]
```diff
@@ -0,0 +1,158 @@
+import puppeteer from 'puppeteer';
```
Comment (Antonio-Laguna) [inline plugins/css-prefers-color-scheme/test/_browser.mjs:1]: Shouldn't this be listed as `devDependencies` of this package?

File: plugins/css-prefers-color-scheme/CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,29 @@
 # Changes to Prefers Color Scheme
 
+### Unreleased (major)
+
+- Breaking : removed old CDN urls
+- Breaking : remove `color-depth` queries fallback
+- Breaking : remove 'no-preference' support as this was dropped from the spec
+- Breaking : remove old global object
```
Comment (Antonio-Laguna) [inline plugins/css-prefers-color-scheme/CHANGELOG.md:8]: Very much nit but keeps consistent with this and all of the other CHANGELOG files

```suggestion
- Breaking: removed old CDN urls
- Breaking: remove `color-depth` queries fallback
- Breaking: remove 'no-preference' support as this was dropped from the spec
- Breaking: remove old global object
```

File: plugins/css-prefers-color-scheme/README.md [hunk 4]
```diff
@@ -1,95 +1,321 @@
...
+		--site-color: #222;
+	}
+}
+```
+
+## Browser
+
+```js
+// initialize prefersColorScheme (applies the current OS color scheme, if available)
+const prefersColorScheme = require('css-prefers-color-scheme/browser')();
```
Comment (Antonio-Laguna) [inline plugins/css-prefers-color-scheme/README.md:125]: Is the immediate invocation correct? Also, should we start advocating for `import`?

File: plugins/css-prefers-color-scheme/CHANGELOG.md (hunk 3 above)
Comment (romainmenke) [reply plugins/css-prefers-color-scheme/CHANGELOG.md]: Good catch!

File: plugins/css-prefers-color-scheme/test/_browser.mjs (hunk 2 above)
Comment (romainmenke) [reply plugins/css-prefers-color-scheme/test/_browser.mjs:1]: yup!

File: plugins/css-prefers-color-scheme/package.json (hunk 1 above)
Comment (romainmenke) [reply plugins/css-prefers-color-scheme/package.json:53]: Going to create a separate issue for this

File: plugins/css-prefers-color-scheme/README.md (hunk 4 above)
Comment (romainmenke) [reply plugins/css-prefers-color-scheme/README.md]: > Is the immediate invocation correct?

It is correct, but I don't think it is very clear. (old docs)

I've updated all these examples to use `import`.
The immediate invocation is now also gone.

**Review by Antonio-Laguna — APPROVED** [review]
Nice one!

**Pull request merged**